File: 739-daily-temperatures/daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] nums) {
        int[] answer = new int[nums.length];
        Stack <Integer> stack = new Stack<>();

        for (int i = nums.length-1; i >= 0; i--){
            while (!stack.isEmpty() && nums[stack.peek()] <= nums[i]){
                stack.pop();
            }
            answer[i] = stack.isEmpty() ? 0:stack.peek()-i;
            stack.push(i);
        }
        return answer;
    }
}

File: 739-daily-temperatures/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, nums: List[int]) -> List[int]:
        res = []
        stack = []

        for i in range(len(nums)-1,-1,-1):
            while stack and nums[stack[-1]] <= nums[i]:
                stack.pop()
            
            res.append(stack[-1]-i if stack else 0)
            stack.append(i)
        return res[::-1]



            
        